publish-mqtt/src/main.rs

```diff
@@ -90,29 +73,45 @@ async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
 
     let local = task::LocalSet::new();
 
-    let bluetooth_handle = local.spawn_local(async move {
-        bluetooth_mainloop(homie).await.unwrap();
+    // Connect to the D-Bus system bus (this is blocking, unfortunately).
+    let (dbus_resource, dbus_conn) = dbus_tokio::connection::new_system_sync()?;
+    let dbus_handle = tokio::spawn(async {
+        let err = dbus_resource.await;
+        Err::<(), anyhow::Error>(anyhow::anyhow!(err))
     });
 
+    let bluetooth_handle =
+        local.run_until(async move { bluetooth_mainloop(homie, dbus_conn).await });
+
     // Poll everything to completion, until the first one bombs out.
-    let res: Result<_, Box<dyn Error + Send + Sync>> = try_join! {
-        // LocalSet finished first. Colour me confused.
-        local.map(|()| Ok(println!("WTF?"))),
+    let res: Result<_, anyhow::Error> = try_join! {
+        // The resource is a task that should be spawned onto a tokio compatible
+        // reactor ASAP. If the resource ever finishes, you lost connection to D-Bus.
+        dbus_handle.map(|res| Ok(res??)),
         // Bluetooth finished first. Convert error and get on with your life.
         bluetooth_handle.map(|res| Ok(res?)),
         // MQTT event loop finished first.
-        mqtt_handle,
+        mqtt_handle.map(|res| Ok(res?)),
     };
     res?;
     Ok(())
 }
 
+#[derive(Debug)]
+enum ConnectionStatus {
+    Unknown,
+    SubscribingFailedOnce,
```

Comment:
- [ ] add docstrings to each of these variants. This is the only variant that's really interesting at the moment (see `connect_start_sensor()` for details)